dandiapi/api/views/zarr.py

```diff
@@ -106,8 +111,14 @@ def create(self, request):
         serializer.is_valid(raise_exception=True)
 
         name = serializer.validated_data['name']
-        zarr_archive: ZarrArchive = ZarrArchive(name=name)
-        zarr_archive.save()
+        dandiset = get_object_or_404(
+            Dandiset.objects.visible_to(request.user), id=serializer.validated_data['dandiset']
+        )
+        zarr_archive: ZarrArchive = ZarrArchive(name=name, dandiset=dandiset)
```

Comment:
Would we need to have a permission check here to ensure the user creating the zarr is an owner on the specified dandiset? As is, it seems that any authenticated (but not necessarily authorized) user could create a zarr archive on any dandiset without one yet.

Reply:
yes.

Reply:
Good catch, fixed